File: blog-server/src/main/java/com/blog/domain/entity/Comment.java
package com.blog.domain.entity;

import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

/**
 * 评论实体
 */
@Data
public class Comment {
    private Long id;
    private Long articleId;
    private Long parentId;
    private Long replyToId;
    private String nickname;
    private String email;
    private String website;
    private String content;
    private String ip;
    private String userAgent;
    private Integer status;
    private LocalDateTime createTime;

    // 关联属性
    private String articleTitle;
    private String replyToNickname;
    private List<Comment> children;
}

File: blog-web/src/api/comment.ts
import request from '@/utils/request'

// 获取文章评论
export function getComments(articleId: number) {
  return request.get(`/web/comments/article/${articleId}`)
}

// 提交评论
export function submitComment(data: {
  articleId: number
  parentId?: number
  replyToId?: number
  nickname: string
  email?: string
  website?: string
  content: string
}) {
  return request.post('/web/comments', data)
}